src/TestableIO.System.IO.Abstractions.TestingHelpers/ProductVersionParser.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace System.IO.Abstractions.TestingHelpers
+{
+    /// <summary>
+    /// Provides functionality to parse a product version string into its major, minor, build, and private parts.
+    /// </summary>
+    public static class ProductVersionParser
```

Comment:
Please mark this class as `internal`, as it should not be required outside of the project.